client/my-sites/importer/site-importer/site-importer-input-pane.jsx

```diff
@@ -0,0 +1,243 @@
+					props.importerStatus.customData.sourceAuthors.forEach( author => {
+						mappingFunction( author, currentUser );
+					} );
+				}, nextProps );
+			} else {
+				defer( props => startMappingAuthors( props.importerStatus.importerId ), nextProps );
+			}
+
+			// Do not continue execution of the function as the rest should be executed on the next update.
+			return;
+		}
+
+		if ( singleAuthorSite && has( this.props, 'importerStatus.customData.sourceAuthors' ) ) {
+			// Authors have been mapped, start the import
+			const oldAuthors = every( this.props.importerStatus.customData.sourceAuthors, 'mappedTo' );
+			const newAuthors = every( nextProps.importerStatus.customData.sourceAuthors, 'mappedTo' );
+
+			if ( oldAuthors === false && newAuthors === true ) {
+				defer( props => {
+					startImporting( props.importerStatus );
+				}, nextProps );
+			}
+		}
+	};
+
+	resetErrors = () => {
+		this.setState( {
+			error: false,
+			errorMessage: '',
+		} );
+	};
+
+	setUrl = event => {
+		this.setState( { siteURLInput: event.target.value } );
+	};
+
+	validateSite = () => {
+		const siteURL = this.state.siteURLInput;
+
+		this.setState( { loading: true }, this.resetErrors );
+
+		wpcom.wpcom.req
+			.get( {
+				path: `/sites/${
+					this.props.site.ID
+				}/site-importer/is-site-importable?site_url=${ siteURL }`,
+				apiNamespace: 'wpcom/v2',
+			} )
+			.then( resp => {
+				this.setState( {
+					importStage: 'importable',
+					importData: {
+						title: resp.site_title,
+						supported: resp.supported_content,
+						unsupported: resp.unsupported_content,
+						favicon: resp.favicon,
+						engine: resp.engine,
+					},
+					loading: false,
+					importSiteURL: siteURL,
+				} );
+			} )
+			.catch( err => {
+				this.setState( {
+					loading: false,
+					error: true,
+					errorMessage: `${ err.message }`,
+				} );
+			} );
+	};
+
+	importSite = () => {
+		this.setState( { loading: true }, this.resetErrors );
+
+		wpcom.wpcom.req
+			.post( {
+				path: `/sites/${ this.props.site.ID }/site-importer/import-site`,
+				apiNamespace: 'wpcom/v2',
+				formData: [
+					[ 'import_status', JSON.stringify( toApi( this.props.importerStatus ) ) ],
+					[ 'site_url', this.state.importSiteURL ],
+				],
+			} )
+			.then( resp => {
+				this.setState( { loading: false } );
+
+				const data = fromApi( resp );
+				const action = finishUpload( this.props.importerStatus.importerId )( data );
+				defer( () => {
+					Dispatcher.handleViewAction( action );
+				} );
+			} )
+			.catch( err => {
+				this.setState( {
+					loading: false,
+					error: true,
+					errorMessage: `${ err.message } (${ err.code })`,
+				} );
+			} );
+	};
+
+	resetImport = () => {
+		this.setState(
+			{
+				loading: false,
+				importStage: 'idle',
+			},
+			this.resetErrors
+		);
+	};
+
+	render() {
+		return (
+			<div className="site-importer__site-importer-pane">
+				{ this.state.importStage === 'idle' && (
```

Comment:
I think a slight improvement here would be to make these state labels constants, so we can compare like:
`this.state.importStage === IDLE_STATE &&` etc.

Reply:
It would be awesome to extract that while refactoring the import section, so we can unify it between all the importers.